From 9f1b68a48c93b01bf4ed61a96dafa7f1ac3ca2a5 Mon Sep 17 00:00:00 2001
From: Wook Song <wook16.song@samsung.com>
Date: Wed, 13 Nov 2019 14:45:42 +0900
Subject: [PATCH 1/3] [CMake] Do not call get_linux_name when LINUX_OS_NAME is
 provided

In order to handle other OS platforms that are not compatible with the
function, get_linux_name, in dependencies.cmake, this patch makes this
function be not invoked when LINUX_OS_NAME is provided from outside.

Signed-off-by: Wook Song <wook16.song@samsung.com>
---
 inference-engine/cmake/dependencies.cmake | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/inference-engine/cmake/dependencies.cmake b/inference-engine/cmake/dependencies.cmake
index 682f2e55..89d0c41f 100644
--- a/inference-engine/cmake/dependencies.cmake
+++ b/inference-engine/cmake/dependencies.cmake
@@ -12,9 +12,11 @@ set_temp_directory(TEMP "${IE_MAIN_SOURCE_DIR}")
 include(ExternalProject)
 
 include(linux_name)
-if(COMMAND get_linux_name)
-    get_linux_name(LINUX_OS_NAME)
-endif()
+if (NOT DEFINED LINUX_OS_NAME)
+    if(COMMAND get_linux_name)
+        get_linux_name(LINUX_OS_NAME)
+    endif()
+endif (NOT DEFINED LINUX_OS_NAME)
 
 if (ENABLE_MYRIAD)
     include(vpu_dependencies)
-- 
2.17.1

